demo/src/main/java/com/orange/ods/demo/ui/components/ComponentSearchScreen.kt

```diff
@@ -0,0 +1,131 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.material.TextField
+import androidx.compose.material.TextFieldDefaults
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Close
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.orange.ods.compose.component.list.OdsListItem
+import com.orange.ods.compose.component.list.OdsListItemIcon
+import com.orange.ods.compose.component.list.OdsListItemIconType
+import com.orange.ods.compose.component.list.iconType
+import com.orange.ods.compose.theme.OdsTheme
+import com.orange.ods.demo.R
+import com.orange.ods.demo.ui.LocalMainTopAppBarManager
+import com.orange.ods.utilities.extension.orElse
+
+@Composable
+fun ComponentSearchScreen(state: MutableState<TextFieldValue>, onComponentClick: (Long) -> Unit) {
+
+    LocalMainTopAppBarManager.current.updateTopAppBarTitle(R.string.navigation_item_search)
+    ComponentList(state = state, onComponentClick)
+}
+
+@Composable
+fun SearchView(state: MutableState<TextFieldValue>) {
+    val focusRequester = remember { FocusRequester() }
+    TextField(
+        value = state.value,
+        onValueChange = { value ->
+            state.value = value
+        },
+        placeholder = {
+            Text(text = stringResource(id = R.string.component_search), color = Color.Gray, fontSize = 18.sp)
+        },
+        modifier = Modifier
+            .fillMaxWidth()
+            .focusRequester(focusRequester),
+        textStyle = TextStyle(color = OdsTheme.colors.onSurface, fontSize = 18.sp),
+        trailingIcon = {
+            IconButton(
+                onClick = {
+                    state.value =
+                        TextFieldValue("")// Remove text from TextField when you press the 'X' icon
+                }
+            ) {
+                if (state.value != TextFieldValue("")) {
+                    Icon(
+                        Icons.Default.Close,
+                        contentDescription = "",
+                        modifier = Modifier
+                            .padding(15.dp)
+                            .size(24.dp)
+                    )
+                }
+            }
+        },
```

Comment:
The `trailingIcon` value should be:

```kotlin
        trailingIcon = if (state.value.text.isNotEmpty()) {
            {
                OdsIconButton(
                    onClick = {
                        state.value = TextFieldValue("") // Remove text from TextField when you press the 'X' icon
                    },
                    imageVector = Icons.Default.Close,
                    contentDescription = stringResource(id = R.string.search_clear)
                )
            }
        } else null,
```

When the component exists in the ODS lib, you should use it like `OdsIconButton` here. 
The `contentDescription` helps the vocal synthesis to say something when the cross is focused. So here, you have to add a string in `strings.xml` to help it saying "Clear search".